ecommerce/src/components/HomePage/BestSeller.jsx

```diff
@@ -1,19 +1,33 @@
 import ProductCard from "../Repetitive/ProductCard";
-import { data } from "../../data/data"
+import React, { useEffect, useState } from 'react';
 
 export default function BestSeller() {
-  const bestSeller = data.productl.filter(product => product.category === "bestSeller");
-  const groupedBestSeller = bestSeller.reduce((acc, product, index) => {
-    const groupIndex = Math.floor(index / 4);
+  const [bestSellerProducts, setBestSellerProducts] = useState([]);
 
-    if (!acc[groupIndex]) {
-      acc[groupIndex] = [];
-    }
-
-    acc[groupIndex].push(product);
-
-    return acc;
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch('https://workintech-fe-ecommerce.onrender.com/products');
+        const data = await response.json();
+  
+        console.log('Fetched data:', data);
+  
+        // Assuming the API returns an array of products
+        // Directly set the state with the fetched data
+        setBestSellerProducts(data.products);
+  
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+  
+    // Fetch data when the component mounts
+    fetchData();
   }, []);
+  
+  const productCards = bestSellerProducts.sort((a, b) => b.sell_count - a.sell_count).slice(0, 8).map((item, index) => {
```

Comment:
Bu kısmı API requesti ile alamadığın için mi böyle yaptın? Postman'da filtreleyip sıralamayı test etmelisin.

Reply:
tabiki denerim hocam evet cevap alamadım çıktı alabilmek için yazdım